{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Concatenating all tools and strains at threshold 0, 10, 100, 1000, and 10K"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import allel\n",
    "import pandas as pd\n",
    "\n",
    "\n",
    "strain=['A_J','AKR_J','BALB_CJ','CBA_J','C3H_HeJ','DBA_2J','LP_J']\n",
    "\n",
    "tools=['BioGraph*',\n",
    "'breakdancer',\n",
    "'clever',\n",
    "'delly',\n",
    "'gasv',\n",
    "'gridss',\n",
    "'indelminer',\n",
    "'lumpexpress',\n",
    "'mistrvar',\n",
    "'pindel',\n",
    "'platypus',\n",
    "'popdel',\n",
    "'rdxplorer',\n",
    "'smoove',\n",
    "'sniffles']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 0 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/angelalu/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py:1745: UserWarning: multiple INFO headers for 'SVLEN'\n",
      "  warnings.warn('multiple INFO headers for %r' % k)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/0t/deletions/nf_0t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/0t/deletions/nf_0t.platypus.BALB_CJ.chr19.100p_sorted.modified.vcf\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1725</td>\n",
       "      <td>FP</td>\n",
       "      <td>6055203</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>78</td>\n",
       "      <td>FP</td>\n",
       "      <td>17072687</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>114</td>\n",
       "      <td>FP</td>\n",
       "      <td>17231867</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J      1725   FP   6055203         0\n",
       "1  sniffles   LP_J        78   FP  17072687         0\n",
       "2  sniffles   LP_J  34582728   FP  17098523         0\n",
       "3  sniffles   LP_J        48   FP  17201451         0\n",
       "4  sniffles   LP_J       114   FP  17231867         0"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#0 threshold deletions\n",
    "\n",
    "th=0\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_0 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_0 = pd.concat([df_current, df_0],ignore_index=True)   \n",
    "        \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "df_0.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [tool, strain, length, flag, position, threshold]\n",
       "Index: []"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#0 threshold non-deletions\n",
    "import os\n",
    "\n",
    "th=0\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_0_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        \n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            print (\"----\")\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_0_nondel = pd.concat([df_current, df_0_nondel],ignore_index=True)   \n",
    "            \n",
    "        #else:\n",
    "            #print(\"Empty File: \" + file)\n",
    "        \n",
    "df_0_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-5-2ce3b51ab1df>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 13\u001b[0;31m         \u001b[0mcallset\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mallel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_vcf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mfields\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'*'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     14\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m             \u001b[0mdf_current\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m'tool'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mt\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'strain'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'length'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/SVLEN'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'flag'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/FLAG'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'position'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/POS'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'threshold'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mth\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36mread_vcf\u001b[0;34m(input, fields, exclude_fields, rename_fields, types, numbers, alt_number, fills, region, tabix, samples, transformers, buffer_size, chunk_length, log)\u001b[0m\n\u001b[1;32m    308\u001b[0m         \u001b[0mnumbers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnumbers\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malt_number\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0malt_number\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer_size\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mbuffer_size\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    309\u001b[0m         \u001b[0mchunk_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mchunk_length\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfills\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfills\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mregion\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mregion\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtabix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtabix\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 310\u001b[0;31m         \u001b[0msamples\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msamples\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtransformers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtransformers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    311\u001b[0m     )\n\u001b[1;32m    312\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36miter_vcf_chunks\u001b[0;34m(input, fields, exclude_fields, types, numbers, alt_number, fills, region, tabix, samples, transformers, buffer_size, chunk_length)\u001b[0m\n\u001b[1;32m   1142\u001b[0m     \u001b[0;31m# setup input stream\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1143\u001b[0m     stream = _setup_input_stream(input=input, region=region, tabix=tabix,\n\u001b[0;32m-> 1144\u001b[0;31m                                  buffer_size=buffer_size)\n\u001b[0m\u001b[1;32m   1145\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1146\u001b[0m     \u001b[0;31m# setup iterator\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36m_setup_input_stream\u001b[0;34m(input, region, tabix, buffer_size)\u001b[0m\n\u001b[1;32m   1065\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1066\u001b[0m         \u001b[0;31m# assume no compression\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1067\u001b[0;31m         \u001b[0mfileobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'rb'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffering\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1068\u001b[0m         \u001b[0mclose\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1069\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf'"
     ]
    }
   ],
   "source": [
    "#10 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=10     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_10 = pd.concat([df_current, df_10],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "df_10.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.gasv.A_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.indelminer.A_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.mistrvar.A_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.popdel.A_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.gasv.AKR_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.lumpexpress.AKR_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.mistrvar.AKR_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.rdxplorer.AKR_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.gasv.BALB_CJ.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.mistrvar.BALB_CJ.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.gasv.CBA_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.mistrvar.CBA_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.gasv.C3H_HeJ.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.indelminer.C3H_HeJ.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.popdel.C3H_HeJ.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.gasv.DBA_2J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.mistrvar.DBA_2J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.rdxplorer.DBA_2J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.gasv.LP_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.indelminer.LP_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.mistrvar.LP_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.rdxplorer.LP_J.chr19.100p_sorted.modified.nondel.vcf\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>28641</td>\n",
       "      <td>TN</td>\n",
       "      <td>22431541</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>19855</td>\n",
       "      <td>TN</td>\n",
       "      <td>26270409</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>40409</td>\n",
       "      <td>TN</td>\n",
       "      <td>28848778</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>167267</td>\n",
       "      <td>TN</td>\n",
       "      <td>32209571</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>40453</td>\n",
       "      <td>TN</td>\n",
       "      <td>50557877</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       tool strain  length flag  position threshold\n",
       "0  sniffles   LP_J   28641   TN  22431541        10\n",
       "1  sniffles   LP_J   19855   TN  26270409        10\n",
       "2  sniffles   LP_J   40409   TN  28848778        10\n",
       "3  sniffles   LP_J  167267   TN  32209571        10\n",
       "4  sniffles   LP_J   40453   TN  50557877        10"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10 threshold non-deletions\n",
    "th=10     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_10_nondel = pd.concat([df_current, df_10_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "df_10_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 100 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-1f4434d15b4a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mt\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtools\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m         \u001b[0mfile\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'../raw_data/mouse/custom_vcf_fulldata/'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mth\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m't/deletions/nf_'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mth\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m't.'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mt\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'.'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'.chr19.100p_sorted.modified.vcf'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m         \u001b[0mcallset\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mallel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_vcf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mfields\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'*'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m             \u001b[0mdf_current\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m'tool'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mt\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'strain'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'length'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/SVLEN'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'flag'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/FLAG'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'position'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/POS'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'threshold'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mth\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36mread_vcf\u001b[0;34m(input, fields, exclude_fields, rename_fields, types, numbers, alt_number, fills, region, tabix, samples, transformers, buffer_size, chunk_length, log)\u001b[0m\n\u001b[1;32m    308\u001b[0m         \u001b[0mnumbers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnumbers\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malt_number\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0malt_number\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer_size\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mbuffer_size\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    309\u001b[0m         \u001b[0mchunk_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mchunk_length\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfills\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfills\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mregion\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mregion\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtabix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtabix\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 310\u001b[0;31m         \u001b[0msamples\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msamples\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtransformers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtransformers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    311\u001b[0m     )\n\u001b[1;32m    312\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36miter_vcf_chunks\u001b[0;34m(input, fields, exclude_fields, types, numbers, alt_number, fills, region, tabix, samples, transformers, buffer_size, chunk_length)\u001b[0m\n\u001b[1;32m   1142\u001b[0m     \u001b[0;31m# setup input stream\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1143\u001b[0m     stream = _setup_input_stream(input=input, region=region, tabix=tabix,\n\u001b[0;32m-> 1144\u001b[0;31m                                  buffer_size=buffer_size)\n\u001b[0m\u001b[1;32m   1145\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1146\u001b[0m     \u001b[0;31m# setup iterator\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36m_setup_input_stream\u001b[0;34m(input, region, tabix, buffer_size)\u001b[0m\n\u001b[1;32m   1065\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1066\u001b[0m         \u001b[0;31m# assume no compression\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1067\u001b[0;31m         \u001b[0mfileobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'rb'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffering\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1068\u001b[0m         \u001b[0mclose\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1069\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf'"
     ]
    }
   ],
   "source": [
    "#100 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_100 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_100 = pd.concat([df_current, df_100],ignore_index=True)\n",
    "        \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "df_100.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/100t/non-deletions/nf_100t.lumpexpress.AKR_J.chr19.100p_sorted.modified.nondel.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/100t/non-deletions/nf_100t.indelminer.LP_J.chr19.100p_sorted.modified.nondel.vcf\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>24683</td>\n",
       "      <td>TN</td>\n",
       "      <td>21449705</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>21785</td>\n",
       "      <td>TN</td>\n",
       "      <td>22409688</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>28641</td>\n",
       "      <td>TN</td>\n",
       "      <td>22431541</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>261746</td>\n",
       "      <td>TN</td>\n",
       "      <td>22797608</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>149268</td>\n",
       "      <td>TN</td>\n",
       "      <td>23059703</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       tool strain  length flag  position threshold\n",
       "0  sniffles   LP_J   24683   TN  21449705       100\n",
       "1  sniffles   LP_J   21785   TN  22409688       100\n",
       "2  sniffles   LP_J   28641   TN  22431541       100\n",
       "3  sniffles   LP_J  261746   TN  22797608       100\n",
       "4  sniffles   LP_J  149268   TN  23059703       100"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#100 threshold non-deletions\n",
    "\n",
    "th=100     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_100_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_100_nondel = pd.concat([df_current, df_100_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "df_100_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1000 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-ce0ff16a3183>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 13\u001b[0;31m         \u001b[0mcallset\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mallel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_vcf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mfields\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'*'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     14\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m             \u001b[0mdf_current\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m'tool'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mt\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'strain'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'length'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/SVLEN'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'flag'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/FLAG'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'position'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/POS'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'threshold'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mth\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36mread_vcf\u001b[0;34m(input, fields, exclude_fields, rename_fields, types, numbers, alt_number, fills, region, tabix, samples, transformers, buffer_size, chunk_length, log)\u001b[0m\n\u001b[1;32m    308\u001b[0m         \u001b[0mnumbers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnumbers\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malt_number\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0malt_number\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer_size\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mbuffer_size\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    309\u001b[0m         \u001b[0mchunk_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mchunk_length\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfills\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfills\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mregion\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mregion\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtabix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtabix\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 310\u001b[0;31m         \u001b[0msamples\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msamples\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtransformers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtransformers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    311\u001b[0m     )\n\u001b[1;32m    312\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36miter_vcf_chunks\u001b[0;34m(input, fields, exclude_fields, types, numbers, alt_number, fills, region, tabix, samples, transformers, buffer_size, chunk_length)\u001b[0m\n\u001b[1;32m   1142\u001b[0m     \u001b[0;31m# setup input stream\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1143\u001b[0m     stream = _setup_input_stream(input=input, region=region, tabix=tabix,\n\u001b[0;32m-> 1144\u001b[0;31m                                  buffer_size=buffer_size)\n\u001b[0m\u001b[1;32m   1145\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1146\u001b[0m     \u001b[0;31m# setup iterator\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36m_setup_input_stream\u001b[0;34m(input, region, tabix, buffer_size)\u001b[0m\n\u001b[1;32m   1065\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1066\u001b[0m         \u001b[0;31m# assume no compression\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1067\u001b[0;31m         \u001b[0mfileobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'rb'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffering\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1068\u001b[0m         \u001b[0mclose\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1069\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf'"
     ]
    }
   ],
   "source": [
    "#1000 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=1000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_1000 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_1000 = pd.concat([df_current, df_1000],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "           \n",
    "df_1000.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/1000t/non-deletions/nf_1000t.lumpexpress.AKR_J.chr19.100p_sorted.modified.nondel.vcf\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>2164</td>\n",
       "      <td>TN</td>\n",
       "      <td>18924303</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>868204</td>\n",
       "      <td>TN</td>\n",
       "      <td>18926808</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>24683</td>\n",
       "      <td>TN</td>\n",
       "      <td>21449705</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>21785</td>\n",
       "      <td>TN</td>\n",
       "      <td>22409688</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>28641</td>\n",
       "      <td>TN</td>\n",
       "      <td>22431541</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       tool strain  length flag  position threshold\n",
       "0  sniffles   LP_J    2164   TN  18924303      1000\n",
       "1  sniffles   LP_J  868204   TN  18926808      1000\n",
       "2  sniffles   LP_J   24683   TN  21449705      1000\n",
       "3  sniffles   LP_J   21785   TN  22409688      1000\n",
       "4  sniffles   LP_J   28641   TN  22431541      1000"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#1000 threshold non-deletions\n",
    "\n",
    "th=1000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_1000_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_1000_nondel = pd.concat([df_current, df_1000_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "        \n",
    "df_1000_nondel.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10000 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../raw_data/mouse/custom_vcf_fulldata/10000t/deletions/nf_10000t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-11-b041771dd43c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 13\u001b[0;31m         \u001b[0mcallset\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mallel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_vcf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mfields\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'*'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     14\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m             \u001b[0mdf_current\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m'tool'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mt\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'strain'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'length'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/SVLEN'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'flag'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/FLAG'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'position'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallset\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'variants/POS'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'threshold'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mth\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36mread_vcf\u001b[0;34m(input, fields, exclude_fields, rename_fields, types, numbers, alt_number, fills, region, tabix, samples, transformers, buffer_size, chunk_length, log)\u001b[0m\n\u001b[1;32m    308\u001b[0m         \u001b[0mnumbers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnumbers\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malt_number\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0malt_number\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer_size\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mbuffer_size\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    309\u001b[0m         \u001b[0mchunk_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mchunk_length\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfills\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfills\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mregion\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mregion\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtabix\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtabix\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 310\u001b[0;31m         \u001b[0msamples\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msamples\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtransformers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtransformers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    311\u001b[0m     )\n\u001b[1;32m    312\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36miter_vcf_chunks\u001b[0;34m(input, fields, exclude_fields, types, numbers, alt_number, fills, region, tabix, samples, transformers, buffer_size, chunk_length)\u001b[0m\n\u001b[1;32m   1142\u001b[0m     \u001b[0;31m# setup input stream\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1143\u001b[0m     stream = _setup_input_stream(input=input, region=region, tabix=tabix,\n\u001b[0;32m-> 1144\u001b[0;31m                                  buffer_size=buffer_size)\n\u001b[0m\u001b[1;32m   1145\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1146\u001b[0m     \u001b[0;31m# setup iterator\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py\u001b[0m in \u001b[0;36m_setup_input_stream\u001b[0;34m(input, region, tabix, buffer_size)\u001b[0m\n\u001b[1;32m   1065\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1066\u001b[0m         \u001b[0;31m# assume no compression\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1067\u001b[0;31m         \u001b[0mfileobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'rb'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffering\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1068\u001b[0m         \u001b[0mclose\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1069\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../raw_data/mouse/custom_vcf_fulldata/10000t/deletions/nf_10000t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf'"
     ]
    }
   ],
   "source": [
    "#10000 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=10000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10000 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_10000 = pd.concat([df_current, df_10000],ignore_index=True)\n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "        \n",
    "df_10000.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10000t/non-deletions/nf_10000t.lumpexpress.AKR_J.chr19.100p_sorted.modified.nondel.vcf\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>2164</td>\n",
       "      <td>TN</td>\n",
       "      <td>18924303</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>868204</td>\n",
       "      <td>TN</td>\n",
       "      <td>18926808</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>37527</td>\n",
       "      <td>TN</td>\n",
       "      <td>20102302</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>24683</td>\n",
       "      <td>TN</td>\n",
       "      <td>21449705</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>69668</td>\n",
       "      <td>TN</td>\n",
       "      <td>22336470</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       tool strain  length flag  position threshold\n",
       "0  sniffles   LP_J    2164   TN  18924303     10000\n",
       "1  sniffles   LP_J  868204   TN  18926808     10000\n",
       "2  sniffles   LP_J   37527   TN  20102302     10000\n",
       "3  sniffles   LP_J   24683   TN  21449705     10000\n",
       "4  sniffles   LP_J   69668   TN  22336470     10000"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10000 threshold non-deletions\n",
    "\n",
    "th=10000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10000_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_10000_nondel = pd.concat([df_current, df_10000_nondel],ignore_index=True)\n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "           \n",
    "        \n",
    "df_10000_nondel.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Concatenate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "#removed df_0 and df_0_nondel for graphing purposes\n",
    "pdList = [df_10, df_100,df_1000,df_10000, df_10_nondel, df_100_nondel,df_1000_nondel,df_10000_nondel]  # List of your dataframes\n",
    "df_cat = pd.concat(pdList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(112, 4)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#output counts for TP, FP, TN rows are not the same as Varuni's; due to missing files?\n",
    "group_data_TP=df_cat[df_cat['flag'] == 'TP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_TP=group_data_TP.rename(columns={\"flag\": \"nTP\"})\n",
    "group_data_TP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(117, 4)"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_FP=df_cat[df_cat['flag'] == 'FP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_FP=group_data_FP.rename(columns={\"flag\": \"nFP\"})\n",
    "group_data_FP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(394, 4)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_TN=df_cat[df_cat['flag'] == 'TN'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_TN=group_data_TN.rename(columns={\"flag\": \"nTN\"})\n",
    "group_data_TN.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(420, 4)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#fill missing combinations with 0s\n",
    "#removed 0 threshold group since it = 0 for all tools\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_TP['tool'] == t) & (group_data_TP['strain'] == s) & (group_data_TP['threshold'] == th)).any()):\n",
    "                group_data_TP = group_data_TP.append({'tool' : t , 'strain' : s,'threshold' : th,'nTP' : 0} , ignore_index=True)\n",
    "group_data_TP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A_J BioGraph* 100\n",
      "A_J breakdancer 100\n",
      "A_J clever 100\n",
      "A_J delly 100\n",
      "A_J gasv 100\n",
      "A_J gridss 100\n",
      "A_J indelminer 100\n",
      "A_J lumpexpress 100\n",
      "A_J mistrvar 100\n",
      "A_J pindel 100\n",
      "A_J platypus 100\n",
      "A_J popdel 100\n",
      "A_J rdxplorer 100\n",
      "A_J smoove 100\n",
      "A_J sniffles 100\n",
      "AKR_J BioGraph* 100\n",
      "AKR_J breakdancer 100\n",
      "AKR_J clever 100\n",
      "AKR_J delly 100\n",
      "AKR_J gasv 100\n",
      "AKR_J gridss 100\n",
      "AKR_J indelminer 100\n",
      "AKR_J lumpexpress 100\n",
      "AKR_J mistrvar 100\n",
      "AKR_J pindel 100\n",
      "AKR_J platypus 100\n",
      "AKR_J popdel 100\n",
      "AKR_J rdxplorer 100\n",
      "AKR_J smoove 100\n",
      "AKR_J sniffles 100\n",
      "BALB_CJ BioGraph* 100\n",
      "BALB_CJ breakdancer 100\n",
      "BALB_CJ clever 100\n",
      "BALB_CJ delly 100\n",
      "BALB_CJ gasv 100\n",
      "BALB_CJ gridss 100\n",
      "BALB_CJ indelminer 100\n",
      "BALB_CJ lumpexpress 100\n",
      "BALB_CJ mistrvar 100\n",
      "BALB_CJ pindel 10\n",
      "BALB_CJ pindel 100\n",
      "BALB_CJ pindel 1000\n",
      "BALB_CJ pindel 10000\n",
      "BALB_CJ platypus 10\n",
      "BALB_CJ platypus 100\n",
      "BALB_CJ platypus 1000\n",
      "BALB_CJ platypus 10000\n",
      "BALB_CJ popdel 10\n",
      "BALB_CJ popdel 100\n",
      "BALB_CJ popdel 1000\n",
      "BALB_CJ popdel 10000\n",
      "BALB_CJ rdxplorer 10\n",
      "BALB_CJ rdxplorer 100\n",
      "BALB_CJ rdxplorer 1000\n",
      "BALB_CJ rdxplorer 10000\n",
      "BALB_CJ smoove 10\n",
      "BALB_CJ smoove 100\n",
      "BALB_CJ smoove 1000\n",
      "BALB_CJ smoove 10000\n",
      "BALB_CJ sniffles 10\n",
      "BALB_CJ sniffles 100\n",
      "BALB_CJ sniffles 1000\n",
      "BALB_CJ sniffles 10000\n",
      "CBA_J BioGraph* 10\n",
      "CBA_J BioGraph* 100\n",
      "CBA_J BioGraph* 1000\n",
      "CBA_J BioGraph* 10000\n",
      "CBA_J breakdancer 10\n",
      "CBA_J breakdancer 100\n",
      "CBA_J breakdancer 1000\n",
      "CBA_J breakdancer 10000\n",
      "CBA_J clever 10\n",
      "CBA_J clever 100\n",
      "CBA_J clever 1000\n",
      "CBA_J clever 10000\n",
      "CBA_J delly 10\n",
      "CBA_J delly 100\n",
      "CBA_J delly 1000\n",
      "CBA_J delly 10000\n",
      "CBA_J gasv 10\n",
      "CBA_J gasv 100\n",
      "CBA_J gasv 1000\n",
      "CBA_J gasv 10000\n",
      "CBA_J gridss 10\n",
      "CBA_J gridss 100\n",
      "CBA_J gridss 1000\n",
      "CBA_J gridss 10000\n",
      "CBA_J indelminer 10\n",
      "CBA_J indelminer 100\n",
      "CBA_J indelminer 1000\n",
      "CBA_J indelminer 10000\n",
      "CBA_J lumpexpress 10\n",
      "CBA_J lumpexpress 100\n",
      "CBA_J lumpexpress 1000\n",
      "CBA_J lumpexpress 10000\n",
      "CBA_J mistrvar 10\n",
      "CBA_J mistrvar 100\n",
      "CBA_J mistrvar 1000\n",
      "CBA_J mistrvar 10000\n",
      "CBA_J pindel 10\n",
      "CBA_J pindel 100\n",
      "CBA_J pindel 1000\n",
      "CBA_J pindel 10000\n",
      "CBA_J platypus 10\n",
      "CBA_J platypus 100\n",
      "CBA_J platypus 1000\n",
      "CBA_J platypus 10000\n",
      "CBA_J popdel 10\n",
      "CBA_J popdel 100\n",
      "CBA_J popdel 1000\n",
      "CBA_J popdel 10000\n",
      "CBA_J rdxplorer 10\n",
      "CBA_J rdxplorer 100\n",
      "CBA_J rdxplorer 1000\n",
      "CBA_J rdxplorer 10000\n",
      "CBA_J smoove 10\n",
      "CBA_J smoove 100\n",
      "CBA_J smoove 1000\n",
      "CBA_J smoove 10000\n",
      "CBA_J sniffles 10\n",
      "CBA_J sniffles 100\n",
      "CBA_J sniffles 1000\n",
      "CBA_J sniffles 10000\n",
      "C3H_HeJ BioGraph* 10\n",
      "C3H_HeJ BioGraph* 100\n",
      "C3H_HeJ BioGraph* 1000\n",
      "C3H_HeJ BioGraph* 10000\n",
      "C3H_HeJ breakdancer 10\n",
      "C3H_HeJ breakdancer 100\n",
      "C3H_HeJ breakdancer 1000\n",
      "C3H_HeJ breakdancer 10000\n",
      "C3H_HeJ clever 10\n",
      "C3H_HeJ clever 100\n",
      "C3H_HeJ clever 1000\n",
      "C3H_HeJ clever 10000\n",
      "C3H_HeJ delly 10\n",
      "C3H_HeJ delly 100\n",
      "C3H_HeJ delly 1000\n",
      "C3H_HeJ delly 10000\n",
      "C3H_HeJ gasv 10\n",
      "C3H_HeJ gasv 100\n",
      "C3H_HeJ gasv 1000\n",
      "C3H_HeJ gasv 10000\n",
      "C3H_HeJ gridss 10\n",
      "C3H_HeJ gridss 100\n",
      "C3H_HeJ gridss 1000\n",
      "C3H_HeJ gridss 10000\n",
      "C3H_HeJ indelminer 10\n",
      "C3H_HeJ indelminer 100\n",
      "C3H_HeJ indelminer 1000\n",
      "C3H_HeJ indelminer 10000\n",
      "C3H_HeJ lumpexpress 10\n",
      "C3H_HeJ lumpexpress 100\n",
      "C3H_HeJ lumpexpress 1000\n",
      "C3H_HeJ lumpexpress 10000\n",
      "C3H_HeJ mistrvar 10\n",
      "C3H_HeJ mistrvar 100\n",
      "C3H_HeJ mistrvar 1000\n",
      "C3H_HeJ mistrvar 10000\n",
      "C3H_HeJ pindel 10\n",
      "C3H_HeJ pindel 100\n",
      "C3H_HeJ pindel 1000\n",
      "C3H_HeJ pindel 10000\n",
      "C3H_HeJ platypus 10\n",
      "C3H_HeJ platypus 100\n",
      "C3H_HeJ platypus 1000\n",
      "C3H_HeJ platypus 10000\n",
      "C3H_HeJ popdel 10\n",
      "C3H_HeJ popdel 100\n",
      "C3H_HeJ popdel 1000\n",
      "C3H_HeJ popdel 10000\n",
      "C3H_HeJ rdxplorer 10\n",
      "C3H_HeJ rdxplorer 100\n",
      "C3H_HeJ rdxplorer 1000\n",
      "C3H_HeJ rdxplorer 10000\n",
      "C3H_HeJ smoove 10\n",
      "C3H_HeJ smoove 100\n",
      "C3H_HeJ smoove 1000\n",
      "C3H_HeJ smoove 10000\n",
      "C3H_HeJ sniffles 10\n",
      "C3H_HeJ sniffles 100\n",
      "C3H_HeJ sniffles 1000\n",
      "C3H_HeJ sniffles 10000\n",
      "DBA_2J BioGraph* 10\n",
      "DBA_2J BioGraph* 100\n",
      "DBA_2J BioGraph* 1000\n",
      "DBA_2J BioGraph* 10000\n",
      "DBA_2J breakdancer 10\n",
      "DBA_2J breakdancer 100\n",
      "DBA_2J breakdancer 1000\n",
      "DBA_2J breakdancer 10000\n",
      "DBA_2J clever 10\n",
      "DBA_2J clever 100\n",
      "DBA_2J clever 1000\n",
      "DBA_2J clever 10000\n",
      "DBA_2J delly 10\n",
      "DBA_2J delly 100\n",
      "DBA_2J delly 1000\n",
      "DBA_2J delly 10000\n",
      "DBA_2J gasv 10\n",
      "DBA_2J gasv 100\n",
      "DBA_2J gasv 1000\n",
      "DBA_2J gasv 10000\n",
      "DBA_2J gridss 10\n",
      "DBA_2J gridss 100\n",
      "DBA_2J gridss 1000\n",
      "DBA_2J gridss 10000\n",
      "DBA_2J indelminer 10\n",
      "DBA_2J indelminer 100\n",
      "DBA_2J indelminer 1000\n",
      "DBA_2J indelminer 10000\n",
      "DBA_2J lumpexpress 10\n",
      "DBA_2J lumpexpress 100\n",
      "DBA_2J lumpexpress 1000\n",
      "DBA_2J lumpexpress 10000\n",
      "DBA_2J mistrvar 10\n",
      "DBA_2J mistrvar 100\n",
      "DBA_2J mistrvar 1000\n",
      "DBA_2J mistrvar 10000\n",
      "DBA_2J pindel 10\n",
      "DBA_2J pindel 100\n",
      "DBA_2J pindel 1000\n",
      "DBA_2J pindel 10000\n",
      "DBA_2J platypus 10\n",
      "DBA_2J platypus 100\n",
      "DBA_2J platypus 1000\n",
      "DBA_2J platypus 10000\n",
      "DBA_2J popdel 10\n",
      "DBA_2J popdel 100\n",
      "DBA_2J popdel 1000\n",
      "DBA_2J popdel 10000\n",
      "DBA_2J rdxplorer 10\n",
      "DBA_2J rdxplorer 100\n",
      "DBA_2J rdxplorer 1000\n",
      "DBA_2J rdxplorer 10000\n",
      "DBA_2J smoove 10\n",
      "DBA_2J smoove 100\n",
      "DBA_2J smoove 1000\n",
      "DBA_2J smoove 10000\n",
      "DBA_2J sniffles 10\n",
      "DBA_2J sniffles 100\n",
      "DBA_2J sniffles 1000\n",
      "DBA_2J sniffles 10000\n",
      "LP_J BioGraph* 10\n",
      "LP_J BioGraph* 100\n",
      "LP_J BioGraph* 1000\n",
      "LP_J BioGraph* 10000\n",
      "LP_J breakdancer 10\n",
      "LP_J breakdancer 100\n",
      "LP_J breakdancer 1000\n",
      "LP_J breakdancer 10000\n",
      "LP_J clever 10\n",
      "LP_J clever 100\n",
      "LP_J clever 1000\n",
      "LP_J clever 10000\n",
      "LP_J delly 10\n",
      "LP_J delly 100\n",
      "LP_J delly 1000\n",
      "LP_J delly 10000\n",
      "LP_J gasv 10\n",
      "LP_J gasv 100\n",
      "LP_J gasv 1000\n",
      "LP_J gasv 10000\n",
      "LP_J gridss 10\n",
      "LP_J gridss 100\n",
      "LP_J gridss 1000\n",
      "LP_J gridss 10000\n",
      "LP_J indelminer 10\n",
      "LP_J indelminer 100\n",
      "LP_J indelminer 1000\n",
      "LP_J indelminer 10000\n",
      "LP_J lumpexpress 10\n",
      "LP_J lumpexpress 100\n",
      "LP_J lumpexpress 1000\n",
      "LP_J lumpexpress 10000\n",
      "LP_J mistrvar 10\n",
      "LP_J mistrvar 100\n",
      "LP_J mistrvar 1000\n",
      "LP_J mistrvar 10000\n",
      "LP_J pindel 10\n",
      "LP_J pindel 100\n",
      "LP_J pindel 1000\n",
      "LP_J pindel 10000\n",
      "LP_J platypus 10\n",
      "LP_J platypus 100\n",
      "LP_J platypus 1000\n",
      "LP_J platypus 10000\n",
      "LP_J popdel 10\n",
      "LP_J popdel 100\n",
      "LP_J popdel 1000\n",
      "LP_J popdel 10000\n",
      "LP_J rdxplorer 10\n",
      "LP_J rdxplorer 100\n",
      "LP_J rdxplorer 1000\n",
      "LP_J rdxplorer 10000\n",
      "LP_J smoove 10\n",
      "LP_J smoove 100\n",
      "LP_J smoove 1000\n",
      "LP_J smoove 10000\n",
      "LP_J sniffles 10\n",
      "LP_J sniffles 100\n",
      "LP_J sniffles 1000\n",
      "LP_J sniffles 10000\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(117, 4)"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_FP['tool'] == t) & (group_data_FP['strain'] == s) & (group_data_FP['threshold'] == th)).any()):\n",
    "                #group_data_FP = group_data_FP.append({'tool' : t , 'strain' : s,'threshold' : th,'nFP' : 0} , ignore_index=True)\n",
    "                print (s,t,th)\n",
    "group_data_FP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(420, 4)"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_TN['tool'] == t) & (group_data_TN['strain'] == s) & (group_data_TN['threshold'] == th)).any()):\n",
    "                group_data_TN = group_data_TN.append({'tool' : t , 'strain' : s,'threshold' : th,'nTN' : 0} , ignore_index=True)\n",
    "                #print (s,t,th)\n",
    "group_data_TN.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(117, 6)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#merge TP and FP\n",
    "df_merge=pd.merge(group_data_TP, group_data_FP,on=['tool','threshold','strain'])\n",
    "df_merge=pd.merge(df_merge, group_data_TN,on=['tool','threshold','strain'])\n",
    "df_merge.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>72</td>\n",
       "      <td>778</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>94</td>\n",
       "      <td>842</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>90</td>\n",
       "      <td>834</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>A_J</td>\n",
       "      <td>304</td>\n",
       "      <td>164</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>294</td>\n",
       "      <td>168</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>A_J</td>\n",
       "      <td>323</td>\n",
       "      <td>145</td>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>317</td>\n",
       "      <td>145</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>18</td>\n",
       "      <td>1054</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool  threshold   strain  nTP   nFP  nTN\n",
       "0    BioGraph*         10    AKR_J   72   778   11\n",
       "1    BioGraph*         10      A_J   94   842   20\n",
       "2    BioGraph*         10  BALB_CJ   90   834   13\n",
       "3    BioGraph*       1000    AKR_J  269   156  122\n",
       "4    BioGraph*       1000      A_J  304   164  150\n",
       "5    BioGraph*       1000  BALB_CJ  294   168  139\n",
       "6    BioGraph*      10000    AKR_J  291   134  173\n",
       "7    BioGraph*      10000      A_J  323   145  198\n",
       "8    BioGraph*      10000  BALB_CJ  317   145  196\n",
       "9  breakdancer         10    AKR_J   18  1054    4"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>A_J</td>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    strain  n_true\n",
       "0    AKR_J     504\n",
       "1      A_J     533\n",
       "2  BALB_CJ     545\n",
       "3  C3H_HeJ     539\n",
       "4    CBA_J     586\n",
       "5   DBA_2J     609\n",
       "6     LP_J     483"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#true SV deletions\n",
    "df_current = pd.DataFrame(columns=['strain','length'])\n",
    "df = pd.DataFrame(columns=['strain','length'])\n",
    "\n",
    "\n",
    "for s in strain:\n",
    "    file='../gold_standard/mouse_vcf/'+s+'_reference.vcf'\n",
    "    callset = allel.read_vcf(file,fields='*')\n",
    "    \n",
    "    df_current = pd.DataFrame({'strain': s, 'length': callset['variants/SVLEN']})\n",
    "    df = pd.concat([df_current, df],ignore_index=True)\n",
    "group_data_true = df.groupby(['strain'],as_index=False).count()\n",
    "group_data_true=group_data_true.rename(columns={\"length\": \"n_true\"})\n",
    "group_data_true"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(117, 7)"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge=pd.merge(df_merge, group_data_true)\n",
    "df_merge.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculating precision, sensitivity, specificity, f-score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "      <th>nFN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>72</td>\n",
       "      <td>778</td>\n",
       "      <td>11</td>\n",
       "      <td>504</td>\n",
       "      <td>432</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>0.084706</td>\n",
       "      <td>0.013942</td>\n",
       "      <td>0.106352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>122</td>\n",
       "      <td>504</td>\n",
       "      <td>235</td>\n",
       "      <td>0.533730</td>\n",
       "      <td>0.632941</td>\n",
       "      <td>0.438849</td>\n",
       "      <td>0.579117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>173</td>\n",
       "      <td>504</td>\n",
       "      <td>213</td>\n",
       "      <td>0.577381</td>\n",
       "      <td>0.684706</td>\n",
       "      <td>0.563518</td>\n",
       "      <td>0.626480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>18</td>\n",
       "      <td>1054</td>\n",
       "      <td>4</td>\n",
       "      <td>504</td>\n",
       "      <td>486</td>\n",
       "      <td>0.035714</td>\n",
       "      <td>0.016791</td>\n",
       "      <td>0.003781</td>\n",
       "      <td>0.022843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>breakdancer</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>311</td>\n",
       "      <td>225</td>\n",
       "      <td>166</td>\n",
       "      <td>504</td>\n",
       "      <td>193</td>\n",
       "      <td>0.617063</td>\n",
       "      <td>0.580224</td>\n",
       "      <td>0.424552</td>\n",
       "      <td>0.598077</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool  threshold strain  nTP   nFP  nTN  n_true  nFN  sensitivity  \\\n",
       "0    BioGraph*         10  AKR_J   72   778   11     504  432     0.142857   \n",
       "1    BioGraph*       1000  AKR_J  269   156  122     504  235     0.533730   \n",
       "2    BioGraph*      10000  AKR_J  291   134  173     504  213     0.577381   \n",
       "3  breakdancer         10  AKR_J   18  1054    4     504  486     0.035714   \n",
       "4  breakdancer       1000  AKR_J  311   225  166     504  193     0.617063   \n",
       "\n",
       "   precision  specificity   f-score  \n",
       "0   0.084706     0.013942  0.106352  \n",
       "1   0.632941     0.438849  0.579117  \n",
       "2   0.684706     0.563518  0.626480  \n",
       "3   0.016791     0.003781  0.022843  \n",
       "4   0.580224     0.424552  0.598077  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge['nFN']=df_merge['n_true']-df_merge['nTP']\n",
    "\n",
    "df_merge['sensitivity']=df_merge['nTP']/df_merge['n_true']\n",
    "df_merge['precision']=df_merge['nTP']/(df_merge['nTP']+df_merge['nFP'])\n",
    "df_merge['specificity']=df_merge['nTN']/(df_merge['nTN']+df_merge['nFP'])\n",
    "df_merge['f-score']=2*(df_merge['sensitivity']*df_merge['precision'])/(df_merge['sensitivity']+df_merge['precision']+0.00000001)\n",
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nFN</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>256</td>\n",
       "      <td>2454</td>\n",
       "      <td>1326</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>867</td>\n",
       "      <td>488</td>\n",
       "      <td>715</td>\n",
       "      <td>411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>931</td>\n",
       "      <td>424</td>\n",
       "      <td>651</td>\n",
       "      <td>567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>56</td>\n",
       "      <td>3192</td>\n",
       "      <td>1526</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>breakdancer</td>\n",
       "      <td>1000</td>\n",
       "      <td>1026</td>\n",
       "      <td>598</td>\n",
       "      <td>556</td>\n",
       "      <td>567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10000</td>\n",
       "      <td>1070</td>\n",
       "      <td>554</td>\n",
       "      <td>512</td>\n",
       "      <td>727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>clever</td>\n",
       "      <td>10</td>\n",
       "      <td>380</td>\n",
       "      <td>21212</td>\n",
       "      <td>1202</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>clever</td>\n",
       "      <td>1000</td>\n",
       "      <td>1244</td>\n",
       "      <td>9552</td>\n",
       "      <td>338</td>\n",
       "      <td>615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>clever</td>\n",
       "      <td>10000</td>\n",
       "      <td>1365</td>\n",
       "      <td>9431</td>\n",
       "      <td>217</td>\n",
       "      <td>859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>delly</td>\n",
       "      <td>10</td>\n",
       "      <td>440</td>\n",
       "      <td>5146</td>\n",
       "      <td>1142</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool  threshold   nTP    nFP   nFN  nTN\n",
       "0    BioGraph*         10   256   2454  1326   44\n",
       "1    BioGraph*       1000   867    488   715  411\n",
       "2    BioGraph*      10000   931    424   651  567\n",
       "3  breakdancer         10    56   3192  1526   13\n",
       "4  breakdancer       1000  1026    598   556  567\n",
       "5  breakdancer      10000  1070    554   512  727\n",
       "6       clever         10   380  21212  1202   80\n",
       "7       clever       1000  1244   9552   338  615\n",
       "8       clever      10000  1365   9431   217  859\n",
       "9        delly         10   440   5146  1142   75"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#count number of TP/FP/TN/FN for each tool at each threshold\n",
    "df_sum=df_merge.groupby(['tool','threshold'],as_index=False)['nTP','nFP','nFN','nTN'].sum()\n",
    "df_sum.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stored 'df_sum' (DataFrame)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "UsageError: Unknown variable 'nTP'\n"
     ]
    }
   ],
   "source": [
    "%store df_sum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
