{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/angelalu/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py:1745: UserWarning: multiple INFO headers for 'SVLEN'\n",
      "  warnings.warn('multiple INFO headers for %r' % k)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1725</td>\n",
       "      <td>FP</td>\n",
       "      <td>6055203</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>78</td>\n",
       "      <td>FP</td>\n",
       "      <td>17072687</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>114</td>\n",
       "      <td>FP</td>\n",
       "      <td>17231867</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J      1725   FP   6055203         0\n",
       "1  sniffles   LP_J        78   FP  17072687         0\n",
       "2  sniffles   LP_J  34582728   FP  17098523         0\n",
       "3  sniffles   LP_J        48   FP  17201451         0\n",
       "4  sniffles   LP_J       114   FP  17231867         0"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#concatenating all tools and strains at threshold 0, 10, 100, 1000, and 10000\n",
    "import os\n",
    "import allel\n",
    "import pandas as pd\n",
    "\n",
    "#missing popdel files, not included in tools list\n",
    "strain=['A_J','AKR_J','BALB_CJ','CBA_J','C3H_HeJ','DBA_2J','LP_J']\n",
    "tools=['BioGraph*',\n",
    "'breakdancer',\n",
    "'clever',\n",
    "'delly',\n",
    "'gasv',\n",
    "'gridss',\n",
    "'indelminer',\n",
    "'lumpexpress',\n",
    "'mistrvar',\n",
    "'pindel',\n",
    "'platypus',\n",
    "'rdxplorer',\n",
    "'smoove',\n",
    "'sniffles']\n",
    "\n",
    "#0 threshold\n",
    "th=0\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_0 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_0 = pd.concat([df_current, df_0],ignore_index=True)   \n",
    "        \n",
    "df_0.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
